src/data/trainers/trainer-config.ts

```diff
@@ -999,13 +999,16 @@ let t = 0;
  * @param postProcess
  */
 export function getRandomPartyMemberFunc(
-  speciesPool: SpeciesId[],
+  speciesPool: (SpeciesId | SpeciesId[])[],
   trainerSlot: TrainerSlot = TrainerSlot.TRAINER,
   ignoreEvolution = false,
   postProcess?: (enemyPokemon: EnemyPokemon) => void,
 ) {
   return (level: number, strength: PartyMemberStrength) => {
-    let species = randSeedItem(speciesPool);
+    let species: SpeciesId | SpeciesId[];
+    do {
+      species = randSeedItem(speciesPool);
+    } while (Array.isArray(species));
```

Comment:
@SirzBenjie I think as-is this can never pick any items that are in a nested array, because if it picks one of the nested arrays it restarts from the entire array instead of picking from within that nested array.
```suggestion
    let species: SpeciesId | SpeciesId[] = speciesPool;
    do {
      species = randSeedItem(species);
    } while (Array.isArray(species));
```

Reply:
That's true, thx!